src/app/clusters/software-diagnostics-server/software-diagnostics-server.cpp

```diff
@@ -158,12 +158,23 @@ bool emberAfSoftwareDiagnosticsClusterResetWatermarksCallback(app::CommandHandle
 
     uint64_t currentHeapUsed;
     CHIP_ERROR err = DeviceLayer::GetDiagnosticDataProvider().GetCurrentHeapUsed(currentHeapUsed);
-    VerifyOrExit(err == CHIP_NO_ERROR, status = EMBER_ZCL_STATUS_FAILURE);
 
-    err = DeviceLayer::GetDiagnosticDataProvider().SetCurrentHeapHighWatermark(currentHeapUsed);
-    VerifyOrExit(err == CHIP_NO_ERROR, status = EMBER_ZCL_STATUS_FAILURE);
+    // Skip SetCurrentHeapHighWatermark if GetCurrentHeapUsed is not implemented
+    if (err != CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE && err != CHIP_ERROR_NOT_IMPLEMENTED)
```

Comment:
But it doesn't really solve the problem. If it's possible to support CurrentHeapHighWatermark + ResetWatermarks without having CurrentHeapUsed attribute, this code will not reset anything, right? Is it then OK to return a success?

Maybe we should rather replace `DiagnosticDataProvider::SetCurrentHeapHighWatermark` with `DiagnosticDataProvider::ResetWatermarks` which would set the current heap watermark to the current usage even if the current usage is not exposed via the attribute.